src/prefect_server/api/states.py

```diff
@@ -20,7 +23,12 @@
 
 @register_api("states.set_flow_run_state")
 async def set_flow_run_state(
-    flow_run_id: str, state: State, version: int = None, agent_id: str = None
+    flow_run_id: str,
+    state: State,
+    version: int = None,
+    agent_id: str = None,
+    flow_concurrency_lock: Semaphore = None,
```

Comment:
An alternate implementation doesn't need to rely on the flow_concurrency_lock or has_been_concurrency_checked parameters. See comments later in this file (`has_been_concurrency_checked`) and in the GraphQL mutation code ( `flow_concurrency_lock`).